src/libs/actions/Task.js

```diff
@@ -260,9 +266,28 @@ function completeTask(taskReportID, taskTitle) {
     ];
 
     // Update optimistic data for parent report action
-    const optimisticParentReportData = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, completedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
-    if (!_.isEmpty(optimisticParentReportData)) {
-        optimisticData.push(optimisticParentReportData);
+    const optimisticDataForParentReportAction = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, completedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
+    if (!_.isEmpty(optimisticDataForParentReportAction)) {
+        optimisticData.push(optimisticDataForParentReportAction);
+    }
+    // Multiple report actions can link to the same child. Both share destination (task parent) and assignee report link to the same report action.
+    // We need to find and update the other parent report action (in assignee report). More info https://github.com/Expensify/App/issues/23920#issuecomment-1663092717
+    const assigneeChatReportID = lodashGet(ReportUtils.getChatByParticipants(taskReport.participantAccountIDs), 'reportID');
+    // We don't need to create optmistic report action for assignee report if assignee and shareDestination are the same
+    if (assigneeChatReportID && assigneeChatReportID !== taskReport.parentReportID) {
+        const clonedParentReportAction = ReportActionsUtils.getParentReportActionInReport(taskReportID, assigneeChatReportID);
+        if (clonedParentReportAction && clonedParentReportAction.reportActionID) {
+            const optimisticDataForClonedParentReportAction = ReportUtils.getOptimisticDataForParentReportAction(
+                taskReportID,
+                completedTaskReportAction.created,
+                CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                assigneeChatReportID,
+                clonedParentReportAction.reportActionID,
+            );
+            if (!_.isEmpty(optimisticDataForClonedParentReportAction)) {
+                optimisticData.push(optimisticDataForClonedParentReportAction);
+            }
+        }
     }
```

Comment:
Can we flatten this? There are two many nested `if` conditions. If needed write a function that returns the "cloned" reportID and parentReportID (assigneeChatReportID and clonedParentReportActionID). This should help remove one nested if condition for better code readability.

Reply:
okay